openshift-knative-operator/pkg/common/api.go

```diff
@@ -0,0 +1,177 @@
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "HorizontalPodAutoscaler" {
+			return nil
+		}
+		if u.GetAPIVersion() != "autoscaling/v2beta2" {
+			return nil
+		}
+		u.SetAPIVersion("autoscaling/v2")
+		return nil
+	}
+}
+
+// SetSecurityContextForAdmissionController set the required pod security context to avoid issues on K8s 1.25+.
+// For more check:  https://connect.redhat.com/en/blog/important-openshift-changes-pod-security-standards
+func SetSecurityContextForAdmissionController() mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		switch u.GetKind() {
+		case "Deployment":
+			deployment := &appsv1.Deployment{}
+			if err := scheme.Scheme.Convert(u, deployment, nil); err != nil {
+				return fmt.Errorf("failed to convert Unstructured to Deployment: %w", err)
+			}
+			obj := deployment
+			podSpec := &deployment.Spec.Template.Spec
+			containers := podSpec.Containers
+			for i := range containers {
+				setPodSecurityContext(&containers[i])
+			}
+			if err := scheme.Scheme.Convert(obj, u, nil); err != nil {
+				return err
+			}
+		case "Job":
+			job := &batchv1.Job{}
+			if err := scheme.Scheme.Convert(u, job, nil); err != nil {
+				return fmt.Errorf("failed to convert Unstructured to Job: %w", err)
+			}
+			obj := job
+			podSpec := &job.Spec.Template.Spec
+			containers := podSpec.Containers
+			for i := range containers {
+				setPodSecurityContext(&containers[i])
+			}
+			if err := scheme.Scheme.Convert(obj, u, nil); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
+func setPodSecurityContext(container *corev1.Container) {
+	if container.SecurityContext == nil {
+		container.SecurityContext = &corev1.SecurityContext{
+			AllowPrivilegeEscalation: ptr.Bool(false),
+			ReadOnlyRootFilesystem:   ptr.Bool(true),
+			RunAsNonRoot:             ptr.Bool(true),
+			Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
+			SeccompProfile:           &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+		}
+	} else {
+		if container.SecurityContext.RunAsNonRoot == nil {
+			container.SecurityContext.RunAsNonRoot = ptr.Bool(true)
+		}
+		if container.SecurityContext.ReadOnlyRootFilesystem == nil {
+			container.SecurityContext.ReadOnlyRootFilesystem = ptr.Bool(true)
+		}
+		if container.SecurityContext.AllowPrivilegeEscalation == nil {
+			container.SecurityContext.AllowPrivilegeEscalation = ptr.Bool(false)
+		}
+		container.SecurityContext.Capabilities = &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}}
+		if container.SecurityContext.SeccompProfile == nil {
+			container.SecurityContext.SeccompProfile = &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault}
+		}
+	}
+}
+
+// CheckMinimumVersion checks if current K8s version we are on is higher than the one passed.
+// If an error is returned then we
+func CheckMinimumVersion(versioner discovery.ServerVersionInterface, version string) error {
```

Comment:
Can we name this `CheckMinimumKubeVersion` ?  (to avoid confusion w/ serverless versions)

Reply:
Sure